crates/eframe/src/native/run.rs

```diff
@@ -791,190 +1016,569 @@ mod glow_integration {
+                                                .current_gl_context
+                                                .take()
+                                                .unwrap()
+                                                .make_not_current()
+                                                .unwrap()
+                                                .make_current(window.gl_surface.as_ref().unwrap())
+                                                .unwrap(),
+                                        );
+
+                                        if !window
+                                            .gl_surface
+                                            .as_ref()
+                                            .unwrap()
+                                            .is_current(glutin.current_gl_context.as_ref().unwrap())
+                                        {
+                                            let builder = &&glutin.builders[&window.pair];
+                                            log::error!("egui::create_viewport_sync with title: `{}` is not created in main thread, try to use wgpu!", builder.title);
+                                        }
+
+                                        egui_glow::painter::clear(
+                                            &c_gl,
+                                            screen_size_in_pixels,
+                                            [0.0, 0.0, 0.0, 0.0],
+                                        );
+
+                                        c_painter.write().paint_and_update_textures(
+                                            screen_size_in_pixels,
+                                            egui_ctx.pixels_per_point(),
+                                            &clipped_primitives,
+                                            &output.textures_delta,
+                                        );
+                                        crate::profile_scope!("swap_buffers");
+                                        let _ = window
+                                            .gl_surface
+                                            .as_ref()
+                                            .expect("failed to get surface to swap buffers")
+                                            .swap_buffers(
+                                                glutin.current_gl_context.as_ref().expect(
+                                                    "failed to get current context to swap buffers",
+                                                ),
+                                            );
+                                        winit_state.handle_platform_output(
+                                            &win,
+                                            egui_ctx,
+                                            output.platform_output,
+                                        );
+                                    } else {
+                                        break 'try_render;
```

Comment:
isn't this just a `return`?

Reply:
That was part of the past implementations!
I got under the radar for me, thx for pointing that out!